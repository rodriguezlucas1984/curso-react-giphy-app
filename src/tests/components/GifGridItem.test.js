import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas en <GifGridItem/>", () => {
  const title = "Dragon Ball GIF by TOEI Animation UK";
  const url =
    "https://media1.giphy.com/media/977YesTjNfQC7vQiph/giphy.gif?cid=7e2b37eceh1dwr34wa0wro9f9mdsas7g20fy1qh5jc7797uk&rid=giphy.gif&ct=g";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test("debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe de tener un pÃ¡rrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(title);
  });
  test("debe de tener la imagen igual al url y alt de las props", () => {
    const img = wrapper.find("img");
    const { src, alt } = img.props();
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test("debe de tener la animate__fadeIn", () => {
    const div = wrapper.find("div");
    const tieneClase = div.hasClass("animate__fadeIn");
    expect(tieneClase).toBe(true);
  });
});
