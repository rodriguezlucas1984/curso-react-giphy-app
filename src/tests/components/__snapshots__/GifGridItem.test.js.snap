// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Dragon Ball GIF by TOEI Animation UK"
    src="https://media1.giphy.com/media/977YesTjNfQC7vQiph/giphy.gif?cid=7e2b37eceh1dwr34wa0wro9f9mdsas7g20fy1qh5jc7797uk&rid=giphy.gif&ct=g"
  />
  <p>
    Dragon Ball GIF by TOEI Animation UK
  </p>
</div>
`;
